app/controllers/requests.rb

```diff
@@ -0,0 +1,89 @@
+class MakersBnb < Sinatra::Base
+
+  get '/requests' do
+    if current_user
+      @users_requests = Request.all(user_id: current_user.id)
+      @received_requests = Request.all(owner_id: current_user.id)
+      @request_display = @users_requests.map do |x|
+        Space.get(x.space_id)
+      end
+      erb :'requests/index'
+    else
+      redirect to '/sessions/new'
+    end
+  end
+
+  post '/requests/new' do
+
+    requested_date = Date.parse(params[:date_requested]) rescue nil
```

Comment:
I'm guessing you're using it because users might enter a random date string that doesn't parse. In this case, you should try to sanitise the input as a guard condition, rather than letting the error flow into your program. What you're doing here is clever, but a guard condition would be neater, and render this unnecessary.